File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    days = []  # 작업에 필요한 일수를 집어 넣는 배열
    n = len(progresses)
    
    # 1. 작업에 필요한 일수 구하기
    for i in range(n):
        if (100 - progresses[i]) % speeds[i] == 0:
            day = (100 - progresses[i]) // speeds[i]
        else:
            day = (100 - progresses[i]) // speeds[i] + 1
        days.append(day)
    
    # 2. 언제 배포 가능한지 구하기
    max_day = days[0]
    cnt = 1
    
    for i in range(1, n):
        if days[i] <= max_day:
            cnt += 1
        else:
            answer.append(cnt)
            cnt = 1
            max_day = days[i]
    
    answer.append(cnt)  # 마지막 남은 작업들에 대한 배포 추가
    
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        int[] days = new int[progresses.length];
        ArrayList<Integer> result = new ArrayList<>();
        
        for (int i = 0; i<progresses.length; i++) {
            days[i] = ((100-progresses[i])%speeds[i] == 0) ? (100-progresses[i])/speeds[i] : (100-progresses[i])/speeds[i] + 1;
        }
        
        int prev = days[0];
        int count = 1;
        
        for (int i = 1; i<days.length; i++) {
            if (days[i]<=prev) {
                count++;
            } else {
                result.add(count);
                count = 1;
                prev = days[i];
            }
        }
        
        result.add(count);
        
        int[] answer = new int[result.size()];
        for (int i = 0; i<result.size(); i++) {
            answer[i] = result.get(i);
        }
        return answer;
    }
}